org.eclipse.xtext/src/org/eclipse/xtext/resource/containers/LiveShadowedAllContainerState.java

```diff
@@ -51,14 +56,84 @@ public IAllContainersState get(IResourceDescriptions localDescriptions, IAllCont
 
 	@Override
 	public Collection<URI> getContainedURIs(String containerHandle) {
-		Set<URI> result = Sets.newLinkedHashSet();
+		Set<URI> localContainedURIs = Sets.newLinkedHashSet();
 		for (IResourceDescription descriptions : localDescriptions.getAllResourceDescriptions()) {
 			String computedHandle = getContainerHandle(descriptions.getURI());
-			if (computedHandle != null && computedHandle.equals(containerHandle))
-				result.add(descriptions.getURI());
+			if (computedHandle != null && computedHandle.equals(containerHandle)) {
+				localContainedURIs.add(descriptions.getURI());
+			}
 		}
-		result.addAll(globalState.getContainedURIs(containerHandle));
-		return result;
+		Collection<URI> globalContainedURIs = globalState.getContainedURIs(containerHandle);
+		return new Collection<URI>() {
```

Comment:
am not sure if there is a better abstract baseclass.

Reply:
I think AbstractCollection will be a little better

Reply:
that one has many of methods implemented we dont want to